### PR TITLE
8265699: (bf) Scopes passed to ScopedMemoryAccess.copy[Swap]Memory in incorrect order

### PR DESCRIPTION
**Pull request by bplb**
Please consider this request to correct what appear to be `jdk.internal.misc.ScopedMemoryAccess$Scope`s passed in incorrect order to the `copyMemory()` and `copySwapMemory()` methods of `jdk.internal.misc.ScopedMemoryAccess` in some of the `java.nio.Buffer` bulk transfer methods.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265699](https://bugs.openjdk.java.net/browse/JDK-8265699): (bf) Scopes passed to ScopedMemoryAccess.copy[Swap]Memory in incorrect order

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3607/head:pull/3607` \
`$ git checkout pull/3607`

Update a local copy of the PR: \
`$ git checkout pull/3607` \
`$ git pull https://git.openjdk.java.net/jdk pull/3607/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3607`

View PR using the GUI difftool: \
`$ git pr show -t 3607`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3607.diff">https://git.openjdk.java.net/jdk/pull/3607.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template [hunk 1]
```diff
@@ -548,7 +548,7 @@ class Direct$Type$Buffer$RW$$BO$
         assert (pos <= lim);
         int rem = (pos <= lim ? lim - pos : 0);
         try {
-            SCOPED_MEMORY_ACCESS.copyMemory(scope(), null, null,
+            SCOPED_MEMORY_ACCESS.copyMemory(scope(), scope(), null,
```
Comment (mcimadamore) [inline src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template:551]: Is this deliberate? same strange to use same scope for both address and target?

**Review by mcimadamore — COMMENTED** [review]
Looks good except for one edit

File: src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template (hunk 1 above)
Comment (bplb) [reply src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template:551]: Yes, it is. In this case the buffer is copying to itself, moving the data down to the beginning of the buffer, position zero.

**Review by mcimadamore — COMMENTED** [review]
Looks good!

**Review by mcimadamore — APPROVED** [review]

**Pull request closed without merging**